contrib/win32/win32compat/win32-utf8.c

```diff
@@ -0,0 +1,50 @@
+/* 
+ * Temporary Windows versions of functions implemented in utf8.c
+ */
+ #include <stdio.h>
+#include <stdarg.h>
+
+int
+vfmprintf(FILE *f, const char *fmt, va_list list) 
+{
+	return vfprintf(f, fmt, list);
+}
+
+int
+mprintf(const char *fmt, ...) 
+{
+	va_list valist;
+	va_start(valist, fmt);
+	vfmprintf(stdout, fmt, valist);
+	va_end(valist);
+	return 0;
+}
+
+int
+fmprintf(FILE *f, const char *fmt, ...) 
+{
+	va_list valist;
+	va_start(valist, fmt);
+	vfmprintf(f, fmt, valist);
+	va_end(valist);
+	return 0;
+}
+
+int
+snmprintf(char *buf, size_t len, int *written, const char *fmt, ...) 
+{
+	int num;
+	va_list valist;
+	va_start(valist, fmt);
+	num = vsnprintf(buf, len, fmt, valist);
+	va_end(valist);
+	*written = num;
+	return 0;
```

Comment:
return the return value of vfmprintf.. it is useful when we want to check the number of characters written...